### PR TITLE
[SYCL] Use 32-bit integers where it's appropriate for matrix instruct…

### PR DESCRIPTION
**Pull request by MrSidims**
…ions

### CODE REVIEW

File: sycl/include/CL/__spirv/spirv_ops.hpp [hunk 1]
```diff
@@ -75,8 +74,8 @@ template <typename T, typename Tp, std::size_t R, std::size_t C,
           __spv::Scope::Flag S = __spv::Scope::Flag::Subgroup>
 extern __DPCPP_SYCL_EXTERNAL void __spirv_JointMatrixStoreCheckedINTEL(
     T *Ptr, __spv::__spirv_JointMatrixINTEL<Tp, R, C, L, S, U> *Object,
-    std::size_t Stride, size_t Height, size_t Width, size_t CoordX,
-    size_t CoordY, __spv::MatrixLayout Layout = L, __spv::Scope::Flag Sc = S,
+    std::size_t Stride, int32_t Height, int32_t Width, int32_t CoordX,
+    int32_t CoordY, __spv::MatrixLayout Layout = L, __spv::Scope::Flag Sc = S,
     int MemOperand = 0);
```
Comment (dkhaldi) [inline sycl/include/CL/__spirv/spirv_ops.hpp:79]: I also prefer using uint32_t per https://github.com/intel-restricted/applications.compilers.llvm-project/pull/20017#discussion_r1562141264 comment

Comment (MrSidims) [reply sycl/include/CL/__spirv/spirv_ops.hpp:79]: To double check (as SPIR-V spec doesn't (and actually can't specify that)), can't CoordX and CoordY be signed?

Comment (dkhaldi) [reply sycl/include/CL/__spirv/spirv_ops.hpp:79]: As a matter of fact, the coordinates can be negative numbers. Thanks for checking.

File: sycl/include/CL/__spirv/spirv_ops.hpp [hunk 2]
```diff
@@ -177,9 +176,8 @@ __spirv_VectorInsertDynamic(__spv::__spirv_JointMatrixINTEL<T, R, C, L, S, U> *,
 
 template <typename T>
 extern __DPCPP_SYCL_EXTERNAL void __spirv_CooperativeMatrixPrefetchINTEL(
-    T *Ptr, std::size_t coordX, std::size_t coordY, std::size_t NumRows,
-    std::size_t NumCols, unsigned int CacheLevel, __spv::MatrixLayout Layout,
-    std::size_t Stride);
+    T *Ptr, int32_t coordX, int32_t coordY, uint32_t NumRows, uint32_t NumCols,
```
Comment (dkhaldi) [inline sycl/include/CL/__spirv/spirv_ops.hpp:179]: coordX and coordY were removed from the spec. Please make sure you remove them in SPIRV as well.

Comment (MrSidims) [reply sycl/include/CL/__spirv/spirv_ops.hpp:179]: Lets update it step by step and keep coordinates in this PR :) Prefetch requires some update starting from the translator and ending up here.

**Review by dkhaldi — APPROVED** [review]
LGTM

**Comment by MrSidims** [comment]
@llvm-reviewers-runtime please take a look

**Review by asudarsa — APPROVED** [review]
LGTM. Thanks

**Comment by MrSidims** [comment]
@llvm-reviewers-runtime @intel/llvm-gatekeepers can we merge this PR?

**Review by steffenlarsen — APPROVED** [review]
LGTM!

**Pull request merged**